info.go

```diff
@@ -199,13 +208,42 @@ func (pi *ProgramInfo) Runtime() (time.Duration, bool) {
 	return time.Duration(0), false
 }
 
+// Instructions returns the 'xlated' instruction stream of the program
+// after it has been verified and rewritten by the kernel. These instructions
+// cannot be loaded back into the kernel as-is, this is mainly used for
+// inspecting loaded programs for troubleshooting, dumping, etc.
+//
+// For example, map accesses are made to reference their kernel map IDs,
+// not the FDs they had when the program was inserted.
+//
+// The first instruction is marked as a symbol using the Program's name.
+//
+// Available from 4.13. Requires CAP_BPF or equivalent.
+func (pi *ProgramInfo) Instructions() (asm.Instructions, error) {
+	// If the calling process is not BPF-capable, the field will be empty.
+	if len(pi.insns) == 0 {
+		return nil, fmt.Errorf("getting xlated instructions: %w", os.ErrPermission)
```

Comment:
This might be a bit misleading on pre 4.13? Maybe return `ErrNotSupported` instead?

Reply:
Agreed, done.